tests/analysis/Decompilation/dynamic_native/linked_list.c

```diff
@@ -0,0 +1,44 @@
+#include <malloc.h>
+#include <stdint.h>
+
+struct Node {
+  int a;
+  struct Node *prev;
+  struct Node *next;
+};
+
+int __attribute__((noinline)) getPrev(struct Node *N) {
+  if (N->prev)
+    return N->prev->a;
+  return 0;
+}
+int __attribute__((noinline)) getNext(struct Node *N) {
+  if (N->next)
+    return N->next->a;
+  return 0;
+}
+
+int __attribute__((noinline)) main(int argc, char **argv) {
+
+  struct Node *First = (struct Node *)malloc(sizeof(struct Node));
+  First->prev = NULL;
+
+  struct Node *Cur = First;
+  for (int i = 0; i < 10; i++) {
+    struct Node *New = (struct Node *)malloc(sizeof(struct Node));
```

Comment:
I know this code will never run, but still, any reason to use `malloc` instead of locals?
If you like `malloc`, free them too!

Reply:
I want to produce pointers to stuff that is not on the stack, so that the results are not polluted by the quality of previous analyses and the stack might be handled in a special way during decompilation. My goal here is to check the results of type system recovery in the most isolated way possible.
If there are better ways I'm open to suggestions.

Reply:
Makes sense. Just add the `free`s then.

Reply:
fixed